app/src/test/java/com/gathering/android/event/putevent/invitation/AddAttendeesViewModelTest.kt

```diff
@@ -0,0 +1,137 @@
+    }
+
+    @Test
+    fun `onOKButtonClicked navigates to AddEvent screen with email list`() {
+        // GIVEN
+        val attendees = "email1,email2,email3"
+        sut.onViewCreated(attendees, addAttendeeNavigator)
+
+        // WHEN
+        sut.onOKButtonClicked()
+
+        // THEN
+        verify { addAttendeeNavigator.navigateToAddEvent(any()) }
+    }
+
+    @Test
+    fun `onOKButtonClicked navigates to AddEvent screen with empty email list`() {
+        // GIVEN
+        val attendees = ""
+        sut.onViewCreated(attendees, addAttendeeNavigator)
+
+        // WHEN
+        sut.onOKButtonClicked()
+
+        // THEN
+        verify { addAttendeeNavigator.navigateToAddEvent(any()) }
+    }
+    @Test
+    fun `onAttendeeRemoveItemClicked removes email from UI state`() {
+        // GIVEN
+        val attendees = "test1@example.com,test2@example.com"
+
+        // WHEN
+        sut.onAttendeeRemoveItemClicked(attendees)
+
+        // THEN
+        Assert.assertFalse(sut.uiState.value.attendeesEmailList.contains("test1@example.com"))
+    }
+
+    @Test
+    fun `onAddAttendeeButtonClicked sets uiState error message when email field is empty`() {
+        // GIVEN
+        val attendees = ""
+
+        // WHEN
+        sut.onAddAttendeeButtonClicked(attendees)
+
+        // THEN
+        Assert.assertEquals(
+            sut.uiState.value.errorMessage, AddAttendeesViewModel.EMAIL_IS_NOT_VALID
+        )
+    }
+
+    @Test
+    fun `onAddAttendeeButtonClicked sets null uiState errorMessage when email is correct`() {
+        // GIVEN
+        val attendees = "test1@example.com"
+
+        // WHEN
+        sut.onAddAttendeeButtonClicked(attendees)
+
+        // THEN
+        Assert.assertEquals(
+            sut.uiState.value.errorMessage, ""
+        )
+    }
+
+    @Test
+    fun `onAttendeeEmailChanged updates state correctly for valid email`() {
+        // GIVEN
+        val attendees = "email1,email2,email3"
+        sut.onViewCreated(attendees, addAttendeeNavigator)
+
+        // WHEN
+        sut.onAttendeeEmailChanged("newEmail@example.com")
+
+        // THEN
+        val currentState = sut.uiState.value
+        Assert.assertEquals("newEmail@example.com", currentState.attendeeEmail)
+        Assert.assertTrue(currentState.addAttendeeButtonEnable)
+        Assert.assertEquals("", currentState.errorMessage)
+    }
+
+    @Test
+    fun `onAttendeeEmailChanged updates state correctly for invalid email`() {
+        // GIVEN
+        val attendees = "email1,email2,email3"
+        sut.onViewCreated(attendees, addAttendeeNavigator)
+
+        // WHEN
+        sut.onAttendeeEmailChanged("invalidEmail")
+
+        // THEN
+        val currentState = sut.uiState.value
+        Assert.assertEquals("invalidEmail", currentState.attendeeEmail)
+        Assert.assertFalse(currentState.addAttendeeButtonEnable)
+        Assert.assertEquals("", currentState.errorMessage)
```

Comment:
to confirm is this expected ? no error message with invalid email ?

Reply:
correct.